x-pack/plugins/security_solution/server/lib/detection_engine/rules/delete_rules.ts

```diff
@@ -14,5 +15,7 @@ export const deleteRules = async ({
   id,
 }: DeleteRuleOptions) => {
   await rulesClient.delete({ id });
-  ruleStatuses.forEach(async (obj) => ruleStatusClient.delete(obj.id));
+  await asyncForEach(ruleStatuses, async (obj) => {
+    await ruleStatusClient.delete(obj.id);
+  });
```

Comment:
I looked this over and discussed it with a few others. We should be removing this soon so I do not think this will be that bad performance wise unless there is a lot to delete.

If we start to see bad performance we will replace this one with a Promise.all() and batch process it in chunks like we do here:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/lib/detection_engine/rules/update_prepacked_rules.ts#L60-L70

But we are giving this and the linter rule a 👍 for the safety first over performance approach. Thanks!